libs/common/domain/src/lib/record/metadata.model.ts

```diff
@@ -58,10 +58,14 @@ export interface BaseRecord {
   keywords: Array<string> // TODO: handle thesaurus and id
   accessConstraints: Array<AccessConstraint>
   useLimitations: Array<string>
+  legalConstraints?: Array<string>
```

Comment:
legal constraints are `accessConstraints` of type `legal`, so no need for a new field I would say

Reply:
Topics and themes are semantically the same thing I would say. In ISO they are in two different places but we can probably append them together in the `themes` array for now. Non-INSPIRE records will get their themes from the `cl_topic` field, whereas INSPIRE records will get them from the `inspireTheme` field.

Reply:
legalConstraints doesn't go to accessConstraints, it go to useLimitations which is a string Array.

Reply:
Here the code : 

private constraintField = (fieldName: string, output, source) => ({
    ...output,
    ...(fieldName.endsWith('UseLimitationObject')
      ? {
          useLimitations: [
            ...(output.useLimitations || []),
            ...selectField<SourceWithUnknownProps[]>(source, fieldName).map(
              (source: SourceWithUnknownProps) =>
                selectTranslatedValue(source, this.lang3)
            ),
          ],
        }
      : {
          accessConstraints: [
            ...(output.accessConstraints || []),
            ...selectField<SourceWithUnknownProps[]>(source, fieldName).map(
              (field) => ({
                text: selectTranslatedValue(field, this.lang3),
                type: this.getConstraintsType(fieldName),
              })
            ),
          ],
        }),
  })

Reply:
@jahow Des nouvelles sur le sujet?

Reply:
@jahow @fgravin i still need an answer about legalConstraints. Should i use accessConstraints type "legal"  then actually it won't works because there no accessContraints for this field but a useLimitations ?

Reply:
@gkeimeHDF sorry for taking so long to answer. It's a tricky subject, I think we can leave it like that and improve it later on. Thanks!